Doc/reference/import.rst

```diff
@@ -596,9 +596,14 @@ the module.
 .. attribute:: __file__
 .. attribute:: __cached__
 
-   ``__file__`` is optional. If set, this attribute's value must be a
-   string.  The import system may opt to leave ``__file__`` unset if it
-   has no semantic meaning (e.g. a module loaded from a database).
+   ``__file__`` is the pathname of the file from which the module
+   was loaded, if it was loaded from a file. The ``__file__`` attribute
+   might be missing for certain types of modules, such as C modules that
+   are statically linked into the interpreter; for extension modules
+   loaded dynamically from a shared library, it is the pathname of the
+   shared libray file. ``__file__`` is optional. If set, this attribute'
+   value must be a string.The import system may opt to leave ``__file__``
+   unset if it has no semantic meaning (e.g. a module loaded from a database).
```

Comment:
This description is kind of mixed. It goes back and forward with the "if present" and "if missing" situations.

Proposed text:

```
``__file__`` is optional (if set, value must be a string). It indicates 
 the pathname of the file from which the module was loaded (if 
loaded from a file), or the pathname of the shared libray file 
for extension modules loaded dynamically from a shared library.
It might be missing for certain types of modules, such as C 
modules that are statically linked into the interpreter, and the
 import system may opt to leave it unset if it has no semantic 
meaning (e.g. a module loaded from a database).
```

Reply:
Thanks for reviewing my PR. Yes, it was a little confusing. I added your proposed text.

Reply:
I have made the requested changes; please review again.